### ISSUE TITLE
Events are not cloned before they are given to the behavior units.

### ISSUE THREAD

**Issue by gallandarakhneorg** [sarl/sarl#483]
In the previous implementation of Janus, each behavior unit has a different instance of event in order to ensure that the change of an event's attribute has no impact on another behavior unit.

This general behavior is no more applied. The same instance is given to all the behavior units, even in different agents.

**Comment by gallandarakhneorg** [comment]
One possible solution could be to enables only read-only access to the attributes of the occurrences.
It could be done based on the current compiler API that tests the pure functions (in event handler guards). The compiler could generates an error or a warning when it is detecting an not-read-only access to the occurrence's attributes.

**Comment by gallandarakhneorg** [comment]
Detecting the different uses of the event's attribute is complex and cannot be done with the current language grammar: a `const` modifier like the one of C++ is missed.

**Comment by gallandarakhneorg** [comment]
An easier way for solving this issue is to create to transform the event to immutable when giving the event is received by the agent.

The SARL api should be refined as:
```java
abstract class Event {
  //...
 abstract Event toImmutable();
}
```

The SARL compiler should generate the code of `toImmutable` in the concrete events.

**Comment by gallandarakhneorg** [comment]
Should be resolved after #547.
